### PR TITLE
MODORDERS-368 fixed update POL with a specified instanceId

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
https://issues.folio.org/browse/MODORDERS-368

## Approach
* added required logic
* unit tests updated

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java [hunk 1]
```diff
@@ -252,4 +255,15 @@ private static Error convertErrorCodesToError(CompositePoLine compPOL, ErrorCode
     }
     return error;
   }
+
+  private static List<ErrorCodes> validatePackagePoLine(CompositePoLine compPOL) {
+     List<ErrorCodes> errors = new ArrayList<>();
+
+     // The quantity of the electronic resources in the cost must be specified
```
Comment (javalkar-varun) [inline src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java]: I think this comment is misleading. This method is not referencing any resources in cost?

File: src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java [hunk 2]
```diff
@@ -252,4 +255,15 @@ private static Error convertErrorCodesToError(CompositePoLine compPOL, ErrorCode
...
+     if (compPOL.getIsPackage() && compPOL.getInstanceId() != null) {
```
Comment (javalkar-varun) [inline src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java]: Sonarcloud is complaining about `getIsPackage()` for NPE. Think it wants  `Boolean.TRUE.equals(compPOL.getIsPackage())` to safeguard against NPE.

File: src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java (hunk 1 above)
Comment (grabsefx) [reply src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java]: Side effect of copy-paste ) 
removed the comment

File: src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java (hunk 2 above)
Comment (grabsefx) [reply src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java]: updated

**Pull request merged**